Source/Android/jni/Achievements.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2003 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <jni.h>
+#include <map>
+#include <thread>
+
+#include "Achievements.h"
+
+#include "Common/Event.h"
+#include "Common/Thread.h"
+#include "Common/Logging/Log.h"
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+
+#include "DiscIO/CompressedBlob.h"
+
+// Global java_vm class
+extern JavaVM* g_java_vm;
+
+namespace Achievements
+{
+// Achievement handling function
+static bool InTheCloset(AchievementTypes type)
+{
+	if (Core::GetState() == Core::CORE_UNINITIALIZED || Core::GetState() == Core::CORE_PAUSE)
+		return false;
+
+	if (DiscIO::IsGCZBlob(SConfig::GetInstance().m_strFilename))
```

Comment:
Calling IsGCZBlob repeatedly is going to generate a bunch of unnecessary disk IO. Something like this would be better:

    if (DVDInterface::VolumeIsValid() && DVDInterface::GetVolume().GetBlobType() == DiscIO::BlobType::GCZ)

...except I don't think that's thread-safe unless you're doing it from the CPU thread. A simpler way to do it would be to just check this once while booting.

Also, why are you only checking for GCZ, not the other compressed blob types?